bookdown/03-optimization-feature-selection.Rmd

```diff
@@ -109,4 +109,171 @@ Work in progress.
 
 ### Wrapper Methods {#fs-wrapper}
 
-Work in progress - via package `r gh_pkg("mlr-org/mlr3fselect")`
+Wrapper feature selection is supported via the `r mlr_pkg("mlr3fselect")` extension package.
+At the heart of `r mlr_pkg("mlr3fselect")` are the R6 classes:
+
+* `r ref("FSelectInstanceSingleCrit")`, `r ref("FSelectInstanceMultiCrit")`: These two classes describe the feature selection problem and store the results.
+* `r ref("FSelector")`: This class is the base class for implementations of feature selection algorithms.
+
+### The `FSelectInstance` Classes {#fs-wrapper-optimization}
+
+The following sub-section examines the feature selection on the `r ref("mlr_tasks_sonar", text = "Pima")` data set which is used to predict whether or not a patient has diabetes. 
+
+```{r 03-optimization-feature-selection-006}
+task = tsk("pima")
+print(task)
+```
+We use the classification tree from `r cran_pkg("rpart")`.
+
+```{r 03-optimization-feature-selection-007}
+learner = lrn("classif.rpart")
+```
+
+Next, we need to specify how to evaluate the performance of the feature subsets.
+For this, we need to choose a `r ref("Resampling", text = "resampling strategy")` and a `r ref("Measure", text = "performance measure")`.
+
+```{r 03-optimization-feature-selection-008}
+hout = rsmp("holdout")
+measure = msr("classif.ce")
+```
+
+Finally, one has to choose the available budget for the feature selection.
+This is done by selecting one of the available `r ref("Terminator", text = "Terminators")`:
+
+* Terminate after a given time (`r ref("TerminatorClockTime")`)
+* Terminate after a given amount of iterations (`r ref("TerminatorEvals")`)
+* Terminate after a specific performance is reached (`r ref("TerminatorPerfReached")`)
+* Terminate when feature selection does not improve (`r ref("TerminatorStagnation")`)
+* A combination of the above in an *ALL* or *ANY* fashion (`r ref("TerminatorCombo")`)
+
+For this short introduction, we specify a budget of 20 evaluations and then put everything together into a `r ref("FSelectInstanceSingleCrit")`:
+
+```{r 03-optimization-feature-selection-009}
+library("mlr3fselect")
+
+evals20 = trm("evals", n_evals = 20)
+
+instance = FSelectInstanceSingleCrit$new(
+  task = task,
+  learner = learner,
+  resampling = hout,
+  measure = measure,
+  terminator = evals20
+)
+instance
+```
+
+To start the feature selecton, we still need to select a feature selection algorithm which are defined via the `r ref("FSelector")` class.
```

Comment:
typo, should be: "To start the feature selection"

Reply:
though would probably read better as: "To start the feature selection, we still need to select an algorithm which are defined via the `r ref("FSelector")` class."